### PR TITLE
Fix tile CLI

### PR DESCRIPTION
**Pull request by eugene123tw**
### Summary

https://github.com/openvinotoolkit/training_extensions/issues/3166

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by harimkang** [comment]
@eugene123tw  I think this should be added to the 2.0.0 branch as well, right?

File: src/otx/engine/utils/auto_configurator.py [hunk 1]
```diff
@@ -347,6 +347,7 @@ def update_ov_subset_pipeline(self, datamodule: OTXDataModule, subset: str = "te
         data_configuration = datamodule.config
         ov_test_config = self._load_default_config(model_name="openvino_model")["data"]["config"][f"{subset}_subset"]
         subset_config = getattr(data_configuration, f"{subset}_subset")
+        subset_config.batch_size = ov_test_config["batch_size"]
```
Comment (harimkang) [inline src/otx/engine/utils/auto_configurator.py:367]: Does `batch_size` also need to be updated here? and if so, could you include the batch size update in the warning message at the bottom of that code?

File: src/otx/engine/engine.py [hunk 2]
```diff
@@ -340,7 +340,7 @@ def test(
         datamodule = datamodule if datamodule is not None else self.datamodule
 
         is_ir_ckpt = Path(str(checkpoint)).suffix in [".xml", ".onnx"]
-        if is_ir_ckpt and not isinstance(model, OVModel):
+        if is_ir_ckpt:
```
Comment (harimkang) [inline src/otx/engine/engine.py]: The only case where `OVModel` is already used is when using `openvino_model.yaml` as config file, as far as I know. In this case, I don't think we need to update the `datamodule` pipeline and `model`. Is there any reason why this syntax should disappear?

Comment (eugene123tw) [reply src/otx/engine/engine.py]: @harimkang  `datamodule` is set with `OTXTileTestDataset` following `otx test --work_dir /tmp/tile/ --config src/otx/recipe/instance_segmentation/openvino_model.yaml --checkpoint /tmp/tile/.latest/export/exported_model.xml`

As Model API has tiling API, there's no need to use OTXTileTestDataset, thus, we should reset the datamodule if model IR is specified. Tbh, I don't quite understand why there's this condition, and I think it should be changed to `is_ir_ckpt and isinstance(model, OVModel)`

Comment (harimkang) [reply src/otx/engine/engine.py]: This is for the natural flow of the engine. After engine.train and export, in order to use the testing ir model, we need to create a new engine by creating an OVModel. This statement eliminates the need to create a new engine by allowing OVModel to be built and used for each task when the checkpoint is xml.

Comment (eugene123tw) [reply src/otx/engine/engine.py]: Thanks for the explanation, I made a new update.

**Review by jaegukhyun — DISMISSED** [review]

File: src/otx/engine/utils/auto_configurator.py (hunk 1 above)
Comment (eugene123tw) [reply src/otx/engine/utils/auto_configurator.py:367]: Done!

**Review by harimkang — CHANGES_REQUESTED** [review]
Please reflect the changes in this PR. https://github.com/openvinotoolkit/training_extensions/pull/3229

**Review by harimkang — APPROVED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Pull request merged**